### ISSUE TITLE
MySQL errors on fresh installation

### ISSUE THREAD

**Issue by pymander** [andris9/mailtrain#72]
I followed the "Download `install.sh` and go" installation technique from the readme file on a fresh, unused VPS. I am using an external SMTP relay. After entering all of the info and double-checking it, I sent a message to a new list. This resulted in a lot of errors in `/var/log/mailtrain.log'. They look like this:

```
ERR! Triggers Error: ER_BAD_FIELD_ERROR: Unknown column 'enabled' in 'where clause'
ERR! Triggers     at Query.Sequence._packetToError (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:51:14)
ERR! Triggers     at Query.ErrorPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Query.js:83:18)
ERR! Triggers     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Triggers     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Triggers     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Triggers     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Triggers     at emitOne (events.js:96:13)
ERR! Triggers     at Socket.emit (events.js:188:7)
ERR! Triggers     at readableAddChunk (_stream_readable.js:172:18)
ERR! Triggers     at Socket.Readable.push (_stream_readable.js:130:10)
ERR! Triggers     --------------------
ERR! Triggers     at Protocol._enqueue (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:141:48)
ERR! Triggers     at PoolConnection.query (/opt/mailtrain/node_modules/mysql/lib/Connection.js:214:25)
ERR! Triggers     at db.getConnection (/opt/mailtrain/services/triggers.js:27:20)
ERR! Triggers     at Handshake.onConnect (/opt/mailtrain/node_modules/mysql/lib/Pool.js:63:7)
ERR! Triggers     at Handshake.Sequence.end (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:85:24)
ERR! Triggers     at Handshake.Sequence.OkPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:94:8)
ERR! Triggers     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Triggers     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Triggers     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Triggers     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Triggers  { Error: ER_BAD_FIELD_ERROR: Unknown column 'enabled' in 'where clause'
ERR! Triggers     at Query.Sequence._packetToError (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:51:14)
ERR! Triggers     at Query.ErrorPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Query.js:83:18)
ERR! Triggers     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Triggers     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Triggers     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Triggers     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Triggers     at emitOne (events.js:96:13)
ERR! Triggers     at Socket.emit (events.js:188:7)
ERR! Triggers     at readableAddChunk (_stream_readable.js:172:18)
ERR! Triggers     at Socket.Readable.push (_stream_readable.js:130:10)
ERR! Triggers     --------------------
ERR! Triggers     at Protocol._enqueue (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:141:48)
ERR! Triggers     at PoolConnection.query (/opt/mailtrain/node_modules/mysql/lib/Connection.js:214:25)
ERR! Triggers     at db.getConnection (/opt/mailtrain/services/triggers.js:27:20)
ERR! Triggers     at Handshake.onConnect (/opt/mailtrain/node_modules/mysql/lib/Pool.js:63:7)
ERR! Triggers     at Handshake.Sequence.end (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:85:24)
ERR! Triggers     at Handshake.Sequence.OkPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:94:8)
ERR! Triggers     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Triggers     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Triggers     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Triggers     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Triggers   code: 'ER_BAD_FIELD_ERROR',
ERR! Triggers   errno: 1054,
ERR! Triggers   sqlState: '42S22',
ERR! Triggers   index: 0 }
ERR! Mail Error: ER_NO_SUCH_TABLE: Table 'mailtrain.queued' doesn't exist
ERR! Mail     at Query.Sequence._packetToError (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:51:14)
ERR! Mail     at Query.ErrorPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Query.js:83:18)
ERR! Mail     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Mail     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Mail     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Mail     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Mail     at emitOne (events.js:96:13)
ERR! Mail     at Socket.emit (events.js:188:7)
ERR! Mail     at readableAddChunk (_stream_readable.js:172:18)
ERR! Mail     at Socket.Readable.push (_stream_readable.js:130:10)
ERR! Mail     --------------------
ERR! Mail     at Protocol._enqueue (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:141:48)
ERR! Mail     at PoolConnection.query (/opt/mailtrain/node_modules/mysql/lib/Connection.js:214:25)
ERR! Mail     at db.getConnection (/opt/mailtrain/services/sender.js:55:24)
ERR! Mail     at Ping.onOperationComplete [as _callback] (/opt/mailtrain/node_modules/mysql/lib/Pool.js:108:5)
ERR! Mail     at Ping.Sequence.end (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:85:24)
ERR! Mail     at Ping.Sequence.OkPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:94:8)
ERR! Mail     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Mail     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Mail     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Mail     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Mail Error: ER_NO_SUCH_TABLE: Table 'mailtrain.queued' doesn't exist
ERR! Mail     at Query.Sequence._packetToError (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:51:14)
ERR! Mail     at Query.ErrorPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Query.js:83:18)
ERR! Mail     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Mail     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Mail     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Mail     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Mail     at emitOne (events.js:96:13)
ERR! Mail     at Socket.emit (events.js:188:7)
ERR! Mail     at readableAddChunk (_stream_readable.js:172:18)
ERR! Mail     at Socket.Readable.push (_stream_readable.js:130:10)
ERR! Mail     --------------------
ERR! Mail     at Protocol._enqueue (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:141:48)
ERR! Mail     at PoolConnection.query (/opt/mailtrain/node_modules/mysql/lib/Connection.js:214:25)
ERR! Mail     at db.getConnection (/opt/mailtrain/services/sender.js:55:24)
ERR! Mail     at Ping.onOperationComplete [as _callback] (/opt/mailtrain/node_modules/mysql/lib/Pool.js:108:5)
ERR! Mail     at Ping.Sequence.end (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:85:24)
ERR! Mail     at Ping.Sequence.OkPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:94:8)
ERR! Mail     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Mail     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Mail     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Mail     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Mail     at emitOne (events.js:96:13)
ERR! Mail     at Socket.emit (events.js:188:7)
ERR! Mail     at readableAddChunk (_stream_readable.js:172:18)
ERR! Mail     at Socket.Readable.push (_stream_readable.js:130:10)
ERR! Mail     --------------------
ERR! Mail     at Protocol._enqueue (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:141:48)
ERR! Mail     at PoolConnection.query (/opt/mailtrain/node_modules/mysql/lib/Connection.js:214:25)
ERR! Mail     at db.getConnection (/opt/mailtrain/services/sender.js:55:24)
ERR! Mail     at Ping.onOperationComplete [as _callback] (/opt/mailtrain/node_modules/mysql/lib/Pool.js:108:5)
ERR! Mail     at Ping.Sequence.end (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:85:24)
ERR! Mail     at Ping.Sequence.OkPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:94:8)
ERR! Mail     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Mail     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Mail     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Mail     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Mail Error: ER_NO_SUCH_TABLE: Table 'mailtrain.queued' doesn't exist
ERR! Mail     at Query.Sequence._packetToError (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:51:14)
ERR! Mail     at Query.ErrorPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Query.js:83:18)
ERR! Mail     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Mail     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Mail     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Mail     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
ERR! Mail     at emitOne (events.js:96:13)
ERR! Mail     at Socket.emit (events.js:188:7)
ERR! Mail     at readableAddChunk (_stream_readable.js:172:18)
ERR! Mail     at Socket.Readable.push (_stream_readable.js:130:10)
ERR! Mail     --------------------
ERR! Mail     at Protocol._enqueue (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:141:48)
ERR! Mail     at PoolConnection.query (/opt/mailtrain/node_modules/mysql/lib/Connection.js:214:25)
ERR! Mail     at db.getConnection (/opt/mailtrain/services/sender.js:55:24)
ERR! Mail     at Ping.onOperationComplete [as _callback] (/opt/mailtrain/node_modules/mysql/lib/Pool.js:108:5)
ERR! Mail     at Ping.Sequence.end (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:85:24)
ERR! Mail     at Ping.Sequence.OkPacket (/opt/mailtrain/node_modules/mysql/lib/protocol/sequences/Sequence.js:94:8)
ERR! Mail     at Protocol._parsePacket (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:280:23)
ERR! Mail     at Parser.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:74:12)
ERR! Mail     at Protocol.write (/opt/mailtrain/node_modules/mysql/lib/protocol/Protocol.js:39:16)
ERR! Mail     at Socket.<anonymous> (/opt/mailtrain/node_modules/mysql/lib/Connection.js:109:28)
```

**Comment by pymander** [comment]
It appears that part of `upgrade-00015.sql` failed. There is no `queued` table, but the schema version has been set to 15 in the `settings` table.

**Comment by andris9** [comment]
Fixed in v1.12.1. The upgrade script was correct but the base sql to upgrade from was conflicting. You should remove the existing invalid installation and start again to get the correct tables.
